src/hypothesis/extra/pandas/impl.py

```diff
@@ -0,0 +1,361 @@
+    # numpy dtype and error if it can't.
+    if isinstance(dtype, np.dtype):
+        return False
+    return dtype == 'category'
+
+
+def build_index(draw, index, min_size, max_size):
+    st.check_valid_interval(
+        min_size, max_size, 'min_size', 'max_size'
+    )
+
+    if index is not None:
+        if not is_sequence(index):
+            raise InvalidArgument(
+                'index=%r was of type %s, but expected a sequence' % (
+                    index, type(index).__name__
+                )
+            )
+
+        if max_size is not None and max_size < len(index):
+            raise InvalidArgument(
+                'Provided index=%r has %d items < max_size=%d' % (
+                    index, len(index), max_size))
+            max_size = len(index)
+        if min_size is not None and min_size > len(index):
+            raise InvalidArgument(
+                'Provided index=%r has %d items > min_size=%d' % (
+                    index, len(index), min_size))
+        return index
+
+    if max_size is None:
+        max_size = min_size + 10
+
+    return hrange(draw(st.integers(min_size, max_size)))
+
+
+def dtype_for_elements_strategy(s):
+    return st.shared(
+        s.map(lambda x: pandas.Series([x]).dtype),
+        key=('hypothesis.extra.pandas.dtype_for_elements_strategy', s),
+    )
+
+
+@st.composite
+def series(
+    draw, elements=None, dtype=None, min_size=0, max_size=None, index=None,
+):
+    """Provides a strategy for producing a pandas.Series.
+
+    Arguments:
+
+        elements: a strategy that will be used to generate the individual
+            values in the series. If None, we will attempt to infer a suitable
+            default from the dtype.
+
+        dtype: the numpy.dtype of the resulting series and may be any value
+            that can be passed to numpy.dtype. If None, will use pandas'
+            standard behaviour to infer it from the type of the elements
+            values. Note that if the type of values that comes out of your
+            elements strategy varies, then so will the resulting dtype of the
+            series.
+
+        min_size: the minimum number of entries in the resulting Series.
+
+        max_size: the maximum number of entries in the resulting Series.
+            If an explicit index is provided then max_size may be at most the
+            length of the index. If an index strategy is provided then whenever
+            the drawn index is too short max_size will merely be reduced.
+
+        index: If not None, an index for the resulting series. If provided,
+            the min_size and max_size must be compatible with the index size
+            but will otherwise be ignored.
+
+    """
+
+    index = build_index(
+        draw, index, min_size, max_size
+    )
+
+    if elements is None and dtype is None:
+        raise InvalidArgument(
+            'series require either an elements strategy or a dtype.'
+        )
+
+    size = len(index)
```

Comment:
Nope; you just get a length-zero Series of the requested dtype.  If dtype is not given; it tries drawing an element and using the dtype of that length-one series to construct the empty one; and if `elements` happens to be None you trigger an internal error.

@DRMacIver, what happens for `series(index=[])``?  What should happen?

Reply:
That would be an error, because at least one of dtype or elements has to be specified. But either `series(index=[], elements=blah)` or `series(index=[], dtype=blah)` would give an empty series of the right dtype.